src/main/java/ticket/test/ticketing/TicketingService.java

```diff
@@ -0,0 +1,70 @@
+package ticket.test.ticketing;
+
+import jakarta.persistence.LockModeType;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.jpa.repository.Lock;
+import org.springframework.stereotype.Service;
+
+import org.springframework.transaction.annotation.Transactional;
+import ticket.test.ticketing.db.Ticket;
+import ticket.test.ticketing.db.TicketRepository;
+
+import java.util.Date;
+import java.util.Optional;
+
+@RequiredArgsConstructor
+@Slf4j
+@Service
+public class TicketingService {
+
+    @Autowired
+    private final TicketRepository ticketRepository;
+
+    /*
+    *  티켓을 예약한다.
+    */
+    @Transactional()
+    public TicketingResponseDto getTicket(TicketingRequestDto request) {
+        Ticket duplicate = ticketRepository.findByUserIdAndShowId(request.getUserId(), request.getShowId());
+        if(duplicate != null) {
+            return new TicketingResponseDto("이미 예약된 공연입니다.", null);
+        }
+        ticketRepository.flush();
+
+        Optional<Ticket> savedSeat = ticketRepository.findByShowIdAndSeatForUpdate(request.getShowId(), request.getSeat());
+        if(savedSeat.isPresent()) {
+            return new TicketingResponseDto("이미 선점된 자리입니다.", null);
+        }
+        ticketRepository.flush();
+
+        Ticket ticket = new Ticket(
+                request.getTicketId(), request.getUserId(), request.getShowId(), request.getSeat(), new Date()
+        );
```

Comment:
request 를 각각 unpacking 하는 것과 dto를 인자로 받고 안에서 unpacking 하는 것의 장단점은 무엇이 있을까요? 여기에서는 왜 unpacking 한 것을 인자로 받는 방식을 사용했나요?

Reply:
개발 당시 선언 차이로 인하여 이런 방식으로 짠 것 같아요.

Reply:
개발 당시 선언 차이가 무슨 뜻인가요?

Reply:
param에 들어가는 Dto랑 DB에 저장하려는 class가 달라서 unpacking 한 뒤 인자를 받는 식으로 작성했었어요

Reply:
requestDTO 와 DB entity class 가 다른 건 이해가 되는데 제 말은 다음과 같이 바꾸면 어떤 차이가 있냐는 뜻이었어요.
```suggestion
        Ticket ticket = new Ticket(
                request, new Date()
        );
```
new Date 도 생성자 안에서 만들 수도 있겠네요